server/src/main/java/org/candlepin/pinsetter/core/PinsetterKernel.java

```diff
@@ -170,6 +171,22 @@ else if (!isClustered()) {
             log.debug("Jobs implemented:" + jobImpls);
             Set<JobKey> jobKeys = scheduler.getJobKeys(jobGroupEquals(CRON_GROUP));
 
+            /*
+             * purge jobs that have been deleted from this version of Candlepin.
+             * This is necessary as we might not even have the Class definition
+             * at classpath, Hence any attempt at fetching the JobDetail by the
+             * Scheduler or JobStatus by the JobCurator will fail.
+             */
+            for (JobKey jobKey : jobKeys) {
+                for (String deletedJob : DELETED_JOBS) {
+                    if (jobKey.getName().contains(deletedJob)) {
```

Comment:
@wottop I could have gone with starts with, but I thought contains is safer.
the job key name / job id is the identifier of the specific job instance, and is usually like 
StatisticHistoryTask-d5d1d055-18bb-4103-a2f8-c08b134a3205